BootloaderCorePkg/BootloaderCorePkg.dsc

```diff
@@ -396,4 +396,9 @@
   *_*_*_CC_FLAGS = -DLITE_PRINT
 !endif
 
-
+  # -O0 results in too big size. Override it to -O1 with lto
+!if $(TARGET) == NOOPT
+  *_GCC5_IA32_CC_FLAGS = -flto -O1
+  *_GCC5_IA32_DLINK_FLAGS = -flto -O1
+  *_VS2015x86_IA32_CC_FLAGS = /Ox

```

Comment:
why only 2015 ?

Reply:
For now, let me override GCC only and remove VS override. For VS, default /Od (disable optimization) will be used.